epochStart/metachain/systemSCs.go

```diff
@@ -232,6 +232,13 @@ func (s *systemSCProcessor) ProcessSystemSmartContract(
 		}
 	}
 
+	if s.flagCorrectNumNodesToStake.IsSet() {
+		err := s.cleanAdditionalQueue()
+		if err != nil {
+			return err
```

Comment:
should we consider this a critical error? Shouldn't we just log.Error the error?

Reply:
it here we have a critical error - the contracts will not work as expected.

Reply:
ok, makes sense. Returning error here will avoid of having an inconsistent state. The metachain will halt and we have to do the fixes so it will resume the operation.